source/development_process/index.rst

```diff
@@ -0,0 +1,306 @@
+at the OSL:
+
+Configuration
+~~~~~~~~~~~~~
+
+We use yaml-based configuration for many of our Django projects. This makes
+the configuration easier to read, and in general allows a slightly greater
+degree of freedom in setting up. To see how this is set up, take a look at
+`What's Fresh's settings.py`_.
+
+.. _What's Fresh's settings.py: https://github.com/osu-cass/whats-fresh-api/blob/master/whats_fresh/settings.py
+
+In general, apps will attempt to load settings from environmental variables.
+This is most commonly used with our Docker-based developer environments. If
+the environment variables are not set, it'll load settings from a yaml config
+file, generally stored at ``/opt/app_name/config/config.yml`` by default. This
+can also be overridden with an environment variable.
+
+Application location
+~~~~~~~~~~~~~~~~~~~~
+
+Applications live in a subdirectory of the project. That is to say, if the project
+is named ``project``, and the app is named ``app``, the directory structure will
+look something like this::
+
+  manage.py
+  project/
+    /project/settings.py
+    /app/models.py
+
+Automated testing with Travis
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Automated testing with Travis CI is an incredibly powerful tool. Travis is a
+continuous integration tool, meaning it's designed to run every time someone
+commits code to a repositoy. In our case, it integrates directly into GitHub
+and runs all of our tests, allowing code reviewers to see if a pull request
+breaks something, or if a merge went horribly, horribly wrong.
+
+To set this up, make a ``.travis.yml`` file in the root of the repo. It should
+look something like this::
+
+  language: python
+  python:
+    - "2.7"
+  # command to install dependencies
+  install:
+    - "python setup.py develop"
+    - "pip install flake8"
+  # command to run tests
+  script:
+    - flake8 working_waterfronts/
+    - django-admin test working_waterfronts --settings="working_waterfronts.settings"
+  addons:
+    postgresql: "9.3"
+  before_script:
+    - psql -c 'create database working_waterfronts;' -U postgres
+    - psql -U postgres -c "create extension postgis;" working_waterfronts
+
+Any command put in the ``before_script`` will be run. This allows us to create
+database and extension needed for the app. The ``script`` is the actual test run
+itself. If either command fails, the "build" will fail, and Travis will report
+that it is not safe to merge.
+
+Using flake8 in Travis forces us to keep our code clean and easy to read.
+
+How to Git Out of a Tight Corner
+--------------------------------
+Git is a powerful tool which is generally easy to use, but sometimes it is
+possible to wind up in a weird state. This section of the document describes
+some intermediate level features of git which may be useful to the developer
+team. Developers may also want to read `tricks for configuring git
+<configuring_git.html>`_.
+
+
+- When and when not to force push
```

Comment:
Always, always, always use the fully qualified push command when force pushing.

That is, do `git push --force <remote> <branchname>` and not `git push --force`.